### ISSUE TITLE
New Jenkins security blocks reports javascript

### ISSUE THREAD

**Issue by myoung34** [gatling/gatling#2889]
[None of the graphs show](http://i.imgur.com/RK16lOo.png)

The error shown in the console is: 

```
Blocked script execution in 'http://ci.../LoadTest/25/gatling/report/appsimulation/source/' because the document's frame is sandboxed and the 'allow-scripts' permission is not set.
```

**Comment by slandelle** [comment]
@myoung34 Thanks for reporting. Could you please build the plugin from [sources](https://github.com/jenkinsci/gatling-plugin) and give it a try?

**Comment by myoung34** [comment]
thanks. will report back tomorrow

**Comment by myoung34** [comment]
Still not fixed, I think it's a deeper issue.
https://issues.jenkins-ci.org/browse/JENKINS-32026
https://wiki.jenkins-ci.org/display/SECURITY/Jenkins+Security+Advisory+2015-12-09

basically:

```
In certain configurations, low privilege users were able to create e.g. HTML files in workspaces and archived artifacts that could result in XSS when accessed by other users. Jenkins now sends Content-Security-Policy headers that enables sandboxing and prohibits script execution by default.
```

**Comment by slandelle** [comment]
So it seems

**Comment by dzieciou** [comment]
Setting the following environment variable helped:

-Dhudson.model.DirectoryBrowserSupport.CSP="sandbox allow-scripts; style-src 'unsafe-inline' *;script-src 'unsafe-inline' *;"

**Comment by myoung34** [comment]
Confirmed this worked...

CentOS jenkins file `/etc/sysconfig/jenkins` now contains

```
JENKINS_JAVA_OPTIONS="-Djava.awt.headless=true -Dhudson.model.DirectoryBrowserSupport.CSP=\"sandbox allow-scripts; style-src 'unsafe-inline' *;script-src 'unsafe-inline' *;\""
```

**Comment by slandelle** [comment]
Instructions added on the plugin page: https://wiki.jenkins-ci.org/display/JENKINS/Gatling+Plugin

**Comment by irfanah** [comment]
thanks it worked

**Comment by iccicci** [comment]
Hi everybody,

I'm having the security problem reported here.
Unfortunately simply adding the parameter -Dhudson.model.DirectoryBrowserSupport.CSP="..." does not makes jenkins changing the security headers.
I'm using jenkins 2.60

Is there any other action I have to do?

Thank you,
iCC

**Comment by ccedric** [comment]
@iccicci The issue should be fixed with the version 1.2.2 of the Jenkins Gatling plugin.

What is your version of the plugin?

If you still experience the issue, can you share us a screenshot and the console error?

**Comment by iccicci** [comment]
Thank you @ccedric for you quick reply.

I think I found the problem by myself: I set the parametr as a jenkins option, once I moved it in java options the three related headers changed as I need.

I think U can forget my post, sorry for the confusion

**Comment by p00j4** [comment]
for me, on chrome 62.xxx  none of the solutions worked. Eventually while exploring, changing chrome settings for the Unsandboxed plugin for my jenkins page, got it working.

**How:** On left of the url bar : 'i' icon → site settings → "Unsandboxed plug-in access" : set it to allow.

chrome://settings/content/siteDetails?site=https://your_jenkins_domain.in

[Jenkins > Configuring Content Security Policy > image2017-12-4 19:12:17.png]


However I'm still evaluating if there is a side effect (invitation to hack). Any pro tips?

**Comment by sjethvani** [comment]
Above chrome hack is not working for me , however I found another one for firefox . 
1) Open about:config 
2) disable`security.csp.enable` flag (set this to `false`)
3) That's it . 
Reload page . It should work now

**Comment by ccedric** [comment]
Hi @sjethvani 
The issue should be fixed with the version 1.2.2 of the plugin.
Which version of the plugin are you using?

**Comment by sjethvani** [comment]
Hey @ccedric  , I am not using gatling plugin in jenkins . Infact I encountered this issue while loading some of my custom HTML pages. While googling , I was redirected to this . Since I found a hack in firefox , I thought to share that among others . Hence commented it over here.
